src/Future.js

```diff
@@ -10,6 +10,16 @@ function jail(handler, f){
   };
 }
 
+function fork(m, reject, resolve) {
```

Comment:
I figured this'd be necessary to preserve backwards compatibility (as well as interoperability). For example it could break existing unit tests where someone mocked a Future by `{fork: sinon.spy()}` or something. I guess we could say that this "is their problem" for not passing the right type. I'll undo this last commit if you want.